drivers/gpio/gpio_intel_socfpga.c

```diff
@@ -0,0 +1,310 @@
+/*
+ * Copyright (C) 2022-2023 Intel Corporation
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT intel_socfpga_gpio
+
+/**
+ * @file
+ * @brief Intel GPIO SoCFPGA Controller Driver
+ *
+ * The GPIO controller on Intel SoCFPGA serves
+ * as GPIO modules. This driver provides
+ * the GPIO functionality.
+ *
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/gpio/gpio_utils.h>
+
+#define LOG_LEVEL CONFIG_GPIO_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(gpio_intel_socfpga);
+
+/* Check if reset property is defined */
+#define GPIO_INTEL_INST_RESET_EN(inst)	DT_NODE_HAS_PROP(DT_DRV_INST(inst), resets)
+#define GPIO_INTEL_CHECK_RESET(inst)	GPIO_INTEL_INST_RESET_EN(inst) ||
+#define GPIO_INTEL_RESET_SUPPORT	(DT_INST_FOREACH_STATUS_OKAY(GPIO_INTEL_CHECK_RESET) 0)
```

Comment:
Please place instance macros near the bottom (a convention in Zephyr) along with other instance macros

Reply:
^^ Please address previous review comments.

Part of the reason that DT instance macros are typically defined at the bottom of the file are so that it's easier to determine if they are actually being used.

It seems to me that the above two macros are unused. They should be removed if that's the case.

Reply:
As the GPIO_INTEL_RESET_SUPPORT macro is used in other part of the driver file, I have created this macro at the beginning of the file.

Reply:
I agree with @cfriedt . You shouldn't be using DT inst macros within the `struct gpio_intel_socfpga_config` or `gpio_init` definitions.

Reply:
@MaureenHelm 
Same driver has to support multiple platform (i.e. agilex7 & agilex5) which is the main factor for implementation like this. Its possible that platform could have reset manager and some platform could not have reset manager support. So we have to check if the reset manager property is defined in a node. If yes, then we should toggle the reset manager pin to reset the peripheral during initialization of the driver.